model/mod_list.py

```diff
@@ -208,5 +326,61 @@ def focusOutEvent(self, e: QFocusEvent) -> None:
         return super().focusOutEvent(e)
 
     def mod_clicked(self, item: QListWidgetItem) -> None:
-        """Method to handle clicking on a row"""
-        self.mod_info_signal.emit(item.data(Qt.UserRole)["packageId"])
+        """
+        Method to handle clicking on a row
+        Look up the mod's data by uuid
+        """
+        self.mod_info_signal.emit(item.data(Qt.UserRole)["uuid"])
+
+    def get_mod_url(self, widget_json_data) -> str:
+        url = ""
+        if (
+            widget_json_data["data_source"] == "workshop"
+        ):  # If the mod was parsed from a Steam mods source
+            if widget_json_data.get("steam_url") and isinstance(
+                widget_json_data["steam_url"], str
+            ):  # If the steam_url exists
+                url = widget_json_data.get("steam_url")  # Use the Steam URL
+            elif widget_json_data.get("url") and isinstance(
+                widget_json_data["url"],
+                str,
+            ):  # Otherwise, check if local metadata url exists
+                url = widget_json_data["url"]  # Use the local metadata url
+            else:  # Otherwise, warn & do nothing
+                logger.warning(f"Unable to get url for mod {widget_package_id}") # TODO: Make warning visible
+        elif (
+            widget_json_data["data_source"] == "local"
+        ):  # If the mod was parsed from a local mods source
+            if widget_json_data.get("url") and isinstance(
+                widget_json_data["url"],
+                str,
+            ):  # If the local metadata url exists
+                url = widget_json_data["url"]  # Use the local metadata url
+            elif widget_json_data.get("steam_url") and isinstance(
+                widget_json_data["steam_url"], str
+            ):  # Otherwise, if the mod has steam_url
+                url = widget_json_data.get("steam_url")  # Use the Steam URL
+            else:  # Otherwise, warn & do nothing
+                logger.warning(f"Unable to get url for mod {widget_package_id}") # TODO: Make warning visible
+        elif (
+            widget_json_data["data_source"] == "expansion"
+        ):  # Otherwise, the mod MUST be an expansion
+            if widget_json_data.get("steam_url") and isinstance(
+                widget_json_data["steam_url"], str
+            ):  # If the steam_url exists
+                url = widget_json_data.get("steam_url")  # Use the Steam URL
+            else:  # Otherwise, warn & do nothing
+                logger.warning(f"Unable to get url for mod {widget_package_id}") # TODO: Make warning visible
+        else:  # ??? Not possible
+            logger.critical(
```

Comment:
I agree. I would like to wait to add more debug prints to the log until we have proper "Log levels" available to the user, so that they don't always have to have  larger prints like these unless they are actually looking for an issue by turning the "DEBUG" mode on in settings (to be implemented idea xD)

Reply:
Fixed